ignite/services/network/networkchain/prepare.go

```diff
@@ -146,6 +146,10 @@ func (c Chain) buildGenesis(
 		}
 	}
 
+	if err := applyParamChanges(genesis, gi.ParamChanges); err != nil {
+		return fmt.Errorf("error applying param changes to genesis: %w", err)
+	}
+
```

Comment:
Each time we call the `genesis.Update` we load the file into the memory. It's better and more performative if we call the update less we can:

```go
	changes := make([]jsonfile.UpdateFileOption, 0)
	for _, pc := range gi.ParamChanges {
		changes = append(
			changes,
			jsonfile.WithKeyValueByte(cosmosgenesis.ParamField(pc.Module, pc.Param), pc.Value),
		)
	}

	// update reward related fields if the testnet is incentivized (with a last block height for reward distribution)
	if lastBlockHeight > 0 {
		changes = append(
			changes,
			jsonfile.WithKeyValue(cosmosgenesis.FieldConsumerChainID, spnChainID),
			jsonfile.WithKeyValueInt(cosmosgenesis.FieldLastBlockHeight, lastBlockHeight),
			jsonfile.WithKeyValue(cosmosgenesis.FieldConsensusTimestamp, rewardsInfo.ConsensusState.Timestamp),
			jsonfile.WithKeyValue(cosmosgenesis.FieldConsensusNextValidatorsHash, rewardsInfo.ConsensusState.NextValidatorsHash),
			jsonfile.WithKeyValue(cosmosgenesis.FieldConsensusRootHash, rewardsInfo.ConsensusState.Root.Hash),
			jsonfile.WithKeyValueInt(cosmosgenesis.FieldConsumerUnbondingPeriod, consumerUnbondingTime),
			jsonfile.WithKeyValueUint(cosmosgenesis.FieldConsumerRevisionHeight, rewardsInfo.RevisionHeight),
		)
	}

	if err := genesis.Update(changes...); err != nil {
		return errors.Wrap(err, "genesis cannot be updated for reward related fields")
	}
```